satpy/readers/ghrsst_l2.py

```diff
@@ -24,21 +27,18 @@
 from satpy.readers.file_handlers import BaseFileHandler
 
 
-class SLSTRL2FileHandler(BaseFileHandler):
-    """File handler for Sentinel-3 SSL L2 netCDF files."""
+class GHRSSTL2FileHandler(BaseFileHandler):
+    """File handler for GHRSST L2 netCDF files."""
 
     def __init__(self, filename, filename_info, filetype_info, engine=None):
-        """Initialize the file handler for Sentinel-3 SSL L2 netCDF data."""
-        super(SLSTRL2FileHandler, self).__init__(filename, filename_info, filetype_info)
+        """Initialize the file handler for GHRSST L2 netCDF data."""
+        super(GHRSSTL2FileHandler, self).__init__(filename, filename_info, filetype_info)
 
         if filename.endswith('tar'):
-            import os
-            import tarfile
-            import tempfile
             with tempfile.TemporaryDirectory() as tempdir:
                 with tarfile.open(name=filename, mode='r') as tf:
                     sst_filename = next((name for name in tf.getnames()
-                                        if name.endswith('nc') and 'GHRSST-SSTskin' in name))
+                                         if name.endswith('nc') and 'GHRSST-SSTskin' in name))
                     tf.extract(sst_filename, tempdir)
```

Comment:
I think you could use `tf.extractfile` (https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractfile) to return a file object directly that `open_dataset` will handle. That way you can skip the temporary directory and unpacking and cleaning up.

Reply:
I will have a look to that, thanks for the suggestion!

Reply:
Done! But unittest coverage lacking.